tests/engine-state/detect-engine-state-01/test.yaml

```diff
@@ -0,0 +1,42 @@
+args:
+- --set stream.midstream=true
+
+checks:
+- filter:
+    count: 1
+    match:
+      event_type: flow
+- filter:
+    count: 1
+    match:
+      event_type: stats
+- filter:
+    count: 1
+    match:
+      event_type: alert
+      alert.signature_id: 1
+- filter:
+    count: 0
+    match:
+      event_type: alert
+      alert.signature_id: 2
+- filter:
+    count: 0
+    match:
+      event_type: alert
+      alert.signature_id: 3
+- filter:
+    count: 0
+    match:
+      event_type: alert
+      alert.signature_id: 4
+- filter:
+    count: 1
+    match:
+      event_type: alert
+      alert.signature_id: 5
+- filter:
+    count: 1
+    match:
+      event_type: alert
+      alert.signature_id: 6
```

Comment:
I think that `DeStateSigTest03` expects that a file is stored. I wonder if we should somehow check that here. If we do, this may be a good use case to check: https://github.com/OISF/suricata-verify/tree/master/tests/filestore-v2.1-forced

Reply:
I was thinking more of this check in the test.yaml file: https://github.com/OISF/suricata-verify/blob/master/tests/filestore-v2.1-forced/test.yaml

Reply:
But I think I need a backup here @inashivb would that check and enabling file-info be enough, or would an actual file be needed?

Reply:
yeah I think filestore would require file to be in the traffic..